orsopy/fileio/base.py

```diff
@@ -33,8 +44,14 @@ def _clean(self):
         :return: Cleaned dictionary
         :rtype: dict
         """
-        return dict((k, v) for (k, v) in self.__dict__.items()
-                    if (v is not None or k not in self.orso_optionals))
+        d = {}
+        for k, v in self.__dict__.items():
+            if (v is not None or k not in self.orso_optionals):
+                if hasattr(v, 'orso_optionals'):
+                    d[k] = v._clean()
+                else:
+                    d[k] = v
+        return d
```

Comment:
btw. why is this a hidden method and called _clean? Isn't it the default dictionary representation?

I would use the name "to_dict" to make it similar to "to_json" or "to_yaml".

Reply:
`to_dict` currently exists as a `@dataclass_json` method. I agree, if we don't use `@dataclass_json` but not if we are.

Reply:
Since we aren't using `@dataclass_json` I have changed `_clean` to `to_dict`.